### DISCUSSION TITLE
how to Upload document file? | چطور باید داکیومنت ارسال کرد؟

### DISCUSSION THREAD

**Discussion by sajjad85gh** [kianahmadian/python-bale-bot#39] [Q&A] [answered]
در اجرای این کد 

```py
@client.listen(EventType.MESSAGE)
async def when_receive_message(message: Message):

    file = open("Data/1.pdf", "rb").read()
    
    return await message.reply_document(document=file)
```

این ارور رو میده:

```error on_message chat_id param must be type of Chat or User```

اگه میشه یه راهنمایی بکنین 
و یک سوال دیگه
exeچطور میشه به ربات فایل 
داد که اونوذخیره کنه؟

**Comment by kian-ahmadian** [comment, marked as answer]
وقت بخیر

## مشکل گزارش شده
**باگ در متد `Bot.send_document` با موفقیت رفع شد. از شما بابت گزارش این مشکل سپاسگزارم! 💛**

*توجه : برای دریافت بروز ترین نسخه از طریق `git` نصب کنید و یا در صورتی که میخواهید از طریق `pypi` آن را نصب کنید منتظر نسخه بعدی کتابخانه باشین!*

## دریافت فایل
برای دریافت فایل (با هر نوع تایپی) میتوانید از طریق کلاس [`bale.Message`](https://docs.python-bale-bot.ir/en/stable/api.html#bale.Message) اقدام کنید. همچنین با کلیک بر روی [اینجا](https://github.com/python-bale-bot/python-bale-bot/blob/master/examples/attachment.py#L16) میتونین مثالی از این عملکرد و ببینید 👀

---
موفق و پیروز باشید

**Comment by sajjad85gh** [comment]
ممنون بابت پاسخ گویی سریع

**Comment by sajjad85gh** [comment]
کتابخونه رو با دستور pip install python-bale-bot -U آپدیت کردم و کد رو تغییر دادم:
```@client.listen(EventType.MESSAGE)
async def when_receive_message(message: Message):

    file = open("Data/1.pdf", "rb").read()
    await message.reply("در حال ارسال...")
    
    return await message.reply_document(document=file, file_name="1.pdf")```
    ولی بازم همون ارور رو میده:
    ```error on_message chat_id param must be type of Chat or User```

**Reply by kian-ahmadian** [reply]
> عزیز این روشی که شما دارین از طریق اون پکیج و نصب میکنین `pypi` هستش.

👀 **کتابخانه ما در لحظه تغییراتی که اعمال میکنه رو در pypi بارگذاری نمیکنه و طی ی مدت 14 الی 30 روزه این روند اتفاق می افته!** علتشم این هستش که در این مدت اتفاقات بیشتری میوفته و صرفا به جهت یک تغییر قرار نیست کل ورژن برای مثال از `2.4.7` به `2.4.8` تغییر کنه! 

---

**🔧 در حال حاضر این مشکل در ورژن `master` ما حل شده، که تنها شما برای دریافت اون به `git` نیازمند هستید. [دانلود گیت](https://git-scm.com/download/)**

بعد از دانلود و ستاپ گیت هم میتونین از طریق دستور زیر در cmd، پکیج و داشته باشین!

```
pip install git+https://github.com/python-bale-bot/python-bale-bot
```

**Reply by sajjad85gh** [reply]
بله ممنون

**Reply by kian-ahmadian** [reply]
**در حال حاضر این مشکل برطرف شده است✨**
